pkg/volume/csi/csi_util.go

```diff
@@ -55,7 +55,7 @@ func getCredentialsFromSecret(k8s kubernetes.Interface, secretRef *api.SecretRef
 // saveVolumeData persists parameter data as json file at the provided location
 func saveVolumeData(dir string, fileName string, data map[string]string) error {
 	dataFilePath := filepath.Join(dir, fileName)
-	klog.V(4).Info(log("saving volume data file [%s]", dataFilePath))
+	klog.V(5).Info(log("saving volume data file [%s]", dataFilePath))
```

Comment:
> When raw block devices mount failed, volume manager would frequently rewrite volData.json
I am not sure changing the log level would fix the issue above.  It would simply hide the log message, but the file would still get generated.

Reply:
In the  original PR,  I add logic to check if vol_data.json changed. However, this would read the file first and may add extra cpu load.

Add cache may completely fix this issue but I am not sure if it is necessary because this only occurs during I develop the csi driver.

Reply:
What is the original PR? So are you saying you added code to check the content of vol_data.json to see if it changed ? Did that fix the additional writes ?

Reply:
Yes，but since we need check the file content, we would add additional reads!